access-grant/src/main/java/com/inrupt/client/accessgrant/AccessDenial.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2023 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.inrupt.client.accessgrant;
+
+import static com.inrupt.client.accessgrant.Utils.*;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.inrupt.client.spi.JsonService;
+import com.inrupt.client.spi.ServiceProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.IOUtils;
+
+/**
+ * An Access Grant abstraction, for use with interacting with Solid resources.
+ */
+public class AccessDenial implements AccessCredential {
+
+    private static final String TYPE = "type";
+    private static final String REVOCATION_LIST_2020_STATUS = "RevocationList2020Status";
+    private static final Set<String> supportedTypes = getSupportedTypes();
+    private static final JsonService jsonService = ServiceProvider.getJsonService();
+
+    private final String credential;
+    private final URI issuer;
+    private final URI identifier;
+    private final Set<String> types;
+    private final Set<String> purposes;
+    private final Set<String> modes;
+    private final Set<URI> resources;
+    private final URI recipient;
+    private final URI creator;
+    private final Instant expiration;
+    private final Status status;
+
+    /**
+     * Read a verifiable presentation as an AccessDenial.
+     *
+     * @param grant the Access Denial serialized as a verifiable presentation
+     */
+    protected AccessDenial(final String grant) throws IOException {
```

Comment:
`grant` local here sounds confusing. Is this a request that's being denied? Or is it a denial that's being wrapped?

Reply:
I'll adjust that parameter name